### PR TITLE
valdate project names in create

### LINKED ISSUE #2283: `flutter create test` should be disallowed
We can't actually create a project named "test" because that name conflicts with `package:test`, which we depend upon.  Several people have hit this mysterious error when creating projects.  We should catch this case in the tool, probably by refusing to use that name and explaining why.

### PR DESCRIPTION
**Pull request by devoncarew**
- validate project names in `create` (fix https://github.com/flutter/flutter/issues/2283)

This in-lines our current deps.

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -164,3 +170,67 @@ String _createProjectIdentifier(String name) {
...
+  'stack_trace',
+  'stream_channel',
+  'string_scanner',
+  'test',
+  'typed_data',
+  'utf',
+  'vector_math',
+  'watcher',
+  'yaml',
+]);
```
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/create.dart]: This list seems hard to maintain.  Maybe we should pare this down to names that folks are likely to hit?  The only case I've actually heard of people hitting is "test".

**Comment by abarth** [comment]
LGTM modulo blacklist.

**Comment by Hixie** [comment]
Can't we determine this list automatically from the .packages file of the flutter package?

**Comment by abarth** [comment]
> Can't we determine this list automatically from the .packages file of the flutter package?

We haven't necessarily computed that on the end-developer's machine.  We used to need to compute it to get the engine revision, but now we store the engine revision explicitly in the repo.

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 1 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/create.dart]: I'll pare this list down.

**Comment by devoncarew** [comment]
```
  'args',
  'async',
  'collection',
  'convert',
  'flutter',
  'html',
  'intl',
  'logging',
  'matcher',
  'mime',
  'path',
  'plugin',
  'pool',
  'test',
  'utf',
  'watcher',
  'yaml'
```

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 2]
```diff
@@ -164,3 +170,34 @@ String _createProjectIdentifier(String name) {
...
+  'matcher',
+  'mime',
+  'path',
+  'plugin',
+  'pool',
+  'test',
+  'utf',
+  'watcher',
+  'yaml'
+]);
```
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/create.dart]: Thanks!  LGTM

**Pull request merged**